functions/APIs/queries.js

```diff
@@ -0,0 +1,97 @@
+const { db } = require('../util/admin');
+const axios = require('axios');
+require('dotenv').config();
+
+/*
+find food banks that are a specific distance from a farm
+GET /queryFoodBanksByDistance
+URL params: locationId, distance (miles)
+success response: array of food bank objects
+*/
+exports.queryFoodBanksByDistance = (request, response) => {
+    db.collection('foodbanks').get()
+    .then((data) => {
+        let foodbanks = [];
+        data.forEach((doc) => {
+            foodbanks.push(doc.data());
+        });
+
+        //calculate distances between farms and food banks
+        function calculateDistances(data) {
+            return new Promise(function(resolve, reject) {
+                axios.get(`https://maps.googleapis.com/maps/api/distancematrix/json?units=imperial&origins=place_id:${request.query.locationId}&destinations=place_id:${data.locationId}&key=${process.env.API_KEY}`)
+                .then((response) => {
+                    let distance = response.data.rows[0].elements[0].distance.text;
+                    if (parseInt(distance.split(" ")[0].replace(/,/g, "")) < request.query.distance) {
```

Comment:
Perhaps some documentation on what the the "/,/g," stuff does/means could be helpful here